bin/publish.ls

```diff
@@ -0,0 +1,150 @@
+require! {
+  'chalk' : {bold, red, cyan, yellow, green, dim}
+  'inquirer'
+  'jsonfile'
+  'semver'
+  'shelljs/global'
+}
+
+
+SUBPROJECTS_TO_PUBLISH =
+  'exocom-mock-javascript'
+  'exocom-server'
+  'exorelay-javascript'
+  'exoservice-javascript'
+
+# exocom-server is used because there is no root package.json
+# all node exocommunication services are in lock-step versioning
+{name, version} = jsonfile.readFileSync './exocom-server/package.json'
+
+if process.argv.length != 3
+  display-help!
+
+level = process.argv[2]
+display-help! if level in ['-h, --help']
+
+target-version = get-target-version version, level
+
+confirm-target-version ->
```

Comment:
This is one of the very few areas where backcalls help make code more readable and are not destroying correctness.

```ls
<- confirm-target-version 
ensure-no-open-files!
...